demos/common/samples/common.hpp

```diff
@@ -1114,11 +1114,14 @@ inline std::size_t getTensorBatch(const InferenceEngine::TensorDesc& desc) {
 inline void showAvailableDevices() {
     InferenceEngine::Core ie;
     std::vector<std::string> devices = ie.GetAvailableDevices();
-
-    std::cout << std::endl;
-    std::cout << "Available target devices:";
-    for (const auto& device : devices) {
-        std::cout << "  " << device;
+    if (!devices.empty()) {
+        std::cout << std::endl;
+        std::cout << "Available target devices:";
+        for (const auto& device : devices) {
+            std::cout << "  " << device;
+        }
+    }
+    else {
+        THROW_IE_EXCEPTION << "Not available any target device to infer on";
```

Comment:
I suggest replacing this message with `"Could not find a device available to infer on"`.

Reply:
yeah, as I have seen the PR https://github.com/opencv/open_model_zoo/pull/331, At least it won't print hard code `HDDL` no matter whether it has or not. And I am OK with the exception message. By the way, I think this PR is an option, you can opt to close it if not that necessary.

Reply:
This PR still has value, since it adds the error message. Are you going to update it?